packages/graphback-runtime-knex/src/KnexDBDataProvider.ts

```diff
@@ -39,26 +39,21 @@ export class KnexDBDataProvider<Type = any> implements GraphbackDataProvider<Typ
     const { idField, data: updateData } = getDatabaseArguments(this.tableMap, data);
 
     //tslint:disable-next-line: await-promise
-    const updateResult = await this.db(this.tableName).update(updateData).where(idField.name, '=', idField.value);
-    if (updateResult === 1) {
-      //tslint:disable-next-line: await-promise
-      const dbResult = await this.db.select(this.getSelectedFields(selectedFields)).from(this.tableName).where(idField.name, '=', idField.value);
-      if (dbResult && dbResult[0]) {
-        return dbResult[0]
-      }
+    const updateResult = await this.db(this.tableName).update(updateData, this.getSelectedFields(selectedFields)).where(idField.name, '=', idField.value);
+    if (updateResult && updateResult[0]) {
+      return updateResult[0];
     }
+
     throw new NoDataError(`Cannot update ${this.tableName}`);
   }
 
   public async delete(data: Partial<Type>, selectedFields?: string[]): Promise<Type> {
     const { idField } = getDatabaseArguments(this.tableMap, data);
 
     //tslint:disable-next-line: await-promise
-    const beforeDelete = await this.db.select(this.getSelectedFields(selectedFields)).from(this.tableName).where(idField.name, '=', idField.value);
```

Comment:
@wtrocki I'd love to know the original reason for two separate queries in the first place - I recall there being a reason so don't want to approve this without input on this

Reply:
This provider history originates from times where we were supporting sqllite etc.